### ISSUE TITLE
Adding "change filament" to layer in Preview Slider does not let you manually change filament.

### ISSUE THREAD

**Issue by mturley777** [bambulab/BambuStudio#4459]
### Bambu Studio Version

1.9.3.50

### Where is the application from?

Bambu Lab Official website

### OS version

Windows 11

### Printer

P!P

### How to reproduce

1) Slice a model
2) In Preview, use slider on right to select a layer.
3) Right click mouse over the slider + and select change filament.
4) Slice and print

### Actual results

Model is printed with currently loaded filament and no attempt is made to allow user to change filament.

### Expected results

Model should print with currently loaded filament up to layer containing the layer change.  Then currently loaded filament should be unloaded and user prompted to load next filament.  Also, the prime tower should be used for this model print if selected in settings.

### Project file & Debug log uploads

[No Filament Chg.zip](https://github.com/user-attachments/files/16134650/No.Filament.Chg.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

**Comment by zhimin-zeng-bambulab** [comment]
@mturley777  Could you describe your problem in more detail and add some screenshots if possible?

**Comment by zhimin-zeng-bambulab** [comment]
@mturley777  Only single-color models are allowed to insert "change filament" in the preview page after slicing.

**Comment by mturley777** [comment]
Sure, on a printer that has no AMS connected,  take any single color project that has been setup with multiple filaments:

![Filament Setup](https://github.com/user-attachments/assets/e0cb09bd-52ee-4634-90fd-478d89c6ac48)

Now slice the project and then add a "change filament' on the layer of your choice:

![Change Filament](https://github.com/user-attachments/assets/866cd229-018e-42d1-9a18-a955fd27dd25)

Now re slice and see that indeed the change filament has been set for the print:

![Change Filament set](https://github.com/user-attachments/assets/b0604d7c-32c0-4fef-b04d-cb7dd6680118)
![Filament Change set 2](https://github.com/user-attachments/assets/54254281-3226-4d46-99ac-1d6180f6de8c)

Now print, and observe that the "change filament" does nothing.  The entire project is printed in the filament that is already loaded in the printer.

**Comment by zhimin-zeng-bambulab** [comment]
@mturley777  Do you want to be able to change colors manually when printing with only external filament?

**Comment by mturley777** [comment]
Yes.  Since the function is available, and you can use it to set a filament change, and it shows in the preview that it is set, then it should work to unload the current filament and prompt the user to load the new filament manually.

**Comment by EricDimitri** [comment]
@zhimin-zeng-bambulab zhimin Adding to this, I was trying to print with TPU, multivolor, so I was using external filament, and did a change color, nothing happened. I had to use PAUSE instead of CHANGE COLOR to change color. The bad thing is that it doesnt generate the PRIME TOWER, and TPU  needs that otherwise it will lick for sure. Please add this as a functionality, and also add Prime tower at least as an option, so we can clean the nozzle and generate a continous flow of fillament right before continueing printing. THanks!!!

**Comment by BillHerb** [comment]
same issue here but as you can see in this screenshot at least his preview tab shows the filament changes color mine doesn't show the black even though at layer 2, I inserted a color change to black it all still shows yellow it's the same bug.  this ticket is over a year old is this going to get fixed sometime soon? Also while your at it how come every time you print something you have to hit the re-sync filament button to grab what colors are in the ams it should remember these.  

![Image](https://github.com/user-attachments/assets/d0dd90f6-ab35-4ec9-a795-65b887c4439e)

![Image](https://github.com/user-attachments/assets/7a72a320-dff4-429d-b626-84049f8a1f3e)

**Comment by BillHerb** [comment]
also to be clear i'm using ams and it still doesn't work a manual review of the exported gcode also reveals there is no code entered to actually tell the printer to change the filament
